### PR TITLE
zfsbackup: init at unstable-2019-03-05

### PR DESCRIPTION
**Pull request by sugar700**
###### Motivation for this change

I'm interested in using zfsbackup on my NixOS server, and figured this may as well be upstream. The installation instructions outright say to use nightly, so that's  what I do - the upstream tags are ancient.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by risicle** [comment]
`nix-review` is happy, macos 10.13, binary executes.

File: pkgs/tools/backup/zfsbackup/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# This file was generated by https://github.com/kamilchm/go2nix v1.3.0
+{ stdenv, buildGoPackage, fetchgit }:
```
Comment (markuskowa) [inline pkgs/tools/backup/zfsbackup/default.nix]:
```suggestion
{ lib, buildGoPackage, fetchgit }:
```

File: pkgs/tools/backup/zfsbackup/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchgit {
+    inherit rev;
+    url = "https://github.com/someone1/zfsbackup-go";
+    sha256 = "0yalsfvzmcnc8yfzm3r5dikqrp57spwa16l7gbzvgqqcz4vlnw3n";
+  };
+
+  goDeps = ./deps.nix;
+
+  meta = {
```
Comment (markuskowa) [inline pkgs/tools/backup/zfsbackup/default.nix]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/backup/zfsbackup/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    description = "Backup ZFS snapshots to cloud storage such as Google, Amazon, Azure, etc";
+    homepage = "https://github.com/someone1/zfsbackup-go";
+    license = stdenv.lib.licenses.mit;
```
Comment (markuskowa) [inline pkgs/tools/backup/zfsbackup/default.nix]:
```suggestion
    license = lib.licenses.mit;
    maintainers = [ lib.maintainers.xfix ];
```

File: pkgs/tools/backup/zfsbackup/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+{ lib, buildGoPackage, fetchgit }:
+
+buildGoPackage rec {
+  pname = "zfsbackup";
+  version = "unstable-2019-03-05";
+  rev = "78fea6e99f0a5a4c8513d3a3d1d45fb6750cfddf";
+
+  goPackagePath = "github.com/someone1/zfsbackup-go";
+
+  src = fetchgit {
```
Comment (markuskowa) [inline pkgs/tools/backup/zfsbackup/default.nix]: Since the sources are on GitHub it should be `FetchFromGitHub` here instead of `fetchgit`.

**Review by ghost — APPROVED** [review]

**Comment by markuskowa** [comment]
@GrahamcOfBorg build zfsbackup

**Pull request merged**